tests/test_physics.py

```diff
@@ -73,3 +73,52 @@ def test_kinematics(sim):
         # check that time is increasing in the world
         assert sim.get_world_time() > prev_time
         prev_time = sim.get_world_time()
+
+
+@pytest.mark.skipif(
+    not osp.exists("data/scene_datasets/habitat-test-scenes/skokloster-castle.glb")
+    or not osp.exists("data/objects/"),
+    reason="Requires the habitat-test-scenes and habitat test objects",
+)
+def test_dynamics(sim):
+    # This test assumes that default.phys_scene_config.json contains "physics simulator": "bullet".
+    # TODO: enable dynamic override of this setting in simulation config structure?
+
+    cfg_settings = examples.settings.default_sim_settings.copy()
+
+    cfg_settings[
+        "scene"
+    ] = "data/scene_datasets/habitat-test-scenes/skokloster-castle.glb"
+    # enable the physics simulator: also clears available actions to no-op
+    cfg_settings["enable_physics"] = True
+    cfg_settings["depth_sensor"] = True
+
+    # test loading the physical scene
+    hab_cfg = examples.settings.make_cfg(cfg_settings)
+    sim.reconfigure(hab_cfg)
+    assert sim.get_physics_object_library_size() > 0
+
+    # test adding an object to the world
+    object_id = sim.add_object(0)
+    assert len(sim.get_existing_object_ids()) > 0
+
+    # TODO: enable and test object MotionType setting
+
+    prev_time = 0.0
+    for _ in range(10):
+        # dynamics:
+        sim.set_translation(np.random.rand(3), object_id)
+
+        sim.apply_force(np.random.rand(3), np.random.rand(3), object_id)
+        # TODO: assertion?
+        sim.apply_torque(np.random.rand(3), object_id)
+        # TODO: assertion?
+
+        # TODO: impulses
+
+        # test getting observation
+        obs = sim.step(random.choice(list(hab_cfg.agents[0].action_space.keys())))
+
+        # check that time is increasing in the world
+        assert sim.get_world_time() > prev_time
+        prev_time = sim.get_world_time()
```

Comment:
Random idea on how to test this -- each rendered frame should be quite different (objects moving around). Is it easy to verify?

Reply:
Yeah, that sounds like one way to test this. Remove the set_translation, make the scene deterministic, spawn the objects within camera view, and then check that they are moving. 

We could also check the state of the objects for change, or try to setup a physical scenario such as an anti-gravity force or something like this.

Reply:
>...try to setup a physical scenario such as an anti-gravity force...
 
I implemented this option with assertions and a deterministic scene.